docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,527 @@
+and then migrating all the plugins over to conda-forge.
+
+* [Adjust outputs for multiple qt backends and menuinst=2](https://github.com/conda-forge/napari-feedstock/pull/32)
+* [Plugin migration search query](https://github.com/conda-forge/staged-recipes/pulls?q=is%3Apr+author%3Agoanpeca+created%3A2022-01-01..2022-05-01)
+  (~200 PRs)
+
+At the time of writing, the plugin migration to conda-forge can be considered 90% done, but a long
+tail of non-trivial packages need to be worked on. This is caused by vendored binaries,
+non-compliant licensing schemes or bad packaging practices. Looking forward, a set of guidelines
+in the plugin development documentation should be added.
+
+### Milestone 2: Building conda-based installers for napari
+
+Implementing the `constructor` workflow on napari was mainly done in a single, long-lived PR, that
+has seen a couple of minor updates in the recent releases:
+
+* Main PR: https://github.com/napari/napari/pull/3378, superseded by
+  https://github.com/napari/napari/pull/3555
+* Other PRs:
+    * https://github.com/napari/napari/pull/3462
+    * https://github.com/napari/napari/pull/4185
+    * https://github.com/napari/napari/pull/4210
+    * https://github.com/napari/napari/pull/4221
+    * https://github.com/napari/napari/pull/4307
+    * https://github.com/napari/napari/pull/4309
+    * https://github.com/napari/napari/pull/4387
+    * https://github.com/napari/napari/pull/4444
+    * https://github.com/napari/napari/pull/4447
+
+
+Adding the missing pieces to `constructor` involves changes in four different projects:
+
+* `conda/constructor` itself, the tool that builds the installer
+* `conda/menuinst`, responsible of creating the shortcuts
+* `conda/conda`, which relies on `menuinst` to delegate the shortcut creation, but this code path
+  was only enabled on Windows so far
+* `conda-forge/conda-standalone-feedstock`, which freezes a `conda` install along with their
+  dependencies and adds a thin layer on top to communicate with the `constructor` installer during
+  the installation process.
+
+The full list of PRs is available on
+[this issue comment](https://github.com/napari/napari/issues/4248#issuecomment-1066574137).
+A complete list of features can be found in the packaging documentation [^napari-packaging-docs].
+
+### Milestone 3: Adding support for conda packages in the plugin manager
+
+Support for conda/mamba handling of plugin installations was implemented in a base
+PR and then extended with different PRs:
+
+* Initial PR: https://github.com/napari/napari/pull/2943
+* Other PRs that improved and extended the funcionality:
+    * https://github.com/napari/napari/pull/4074
+    * https://github.com/napari/napari/pull/3369
+    * https://github.com/napari/napari/pull/3288
+
+Adding the necessary functionality to the plugin dialog, including:
+
+* Adapting the current `Installer` class to support not only `pip` but also `conda/mamba` in a
+  single code base.
```

Comment:
I'm of half a mind to make the installer *only* work in the bundle, and leave Python users to install locally using their preferred method. What do you think about this @jaimergp? @napari/core-devs?

Reply:
No. Searching plugins in the application is a really nice feature.

Reply:
I also find it useful!

Reply:
@Czaki @brisvag I have certainly used it, but from a maintenance perspective, if we accept the NAP, the fact that we need to support multiple install mechanisms (conda for bundle, pip for anything else, but maybe conda if it's a conda package?), combined with the fact that the installation chosen by us might break the environment, means that it would be much much simpler and cleaner to not support this. I think the hub presents plugins much more nicely anyway, with gifs etc. Anyway, that is a side issue not related to this NAP, let's keep discussing elsewhere/in a new NAP.

Reply:
> combined with the fact that the installation chosen by us might break the environment, means that it would be much much simpler and cleaner to not support this. I think the hub presents plugins much more nicely anyway, with gifs etc. Anyway, that is a side issue not related to this NAP, let's keep discussing elsewhere/in a new NAP.

I also share this perspective, but I don't think we need to decide this right now

Reply:
probably a bit biased having added the original implementation, but I'm with @Czaki  and @brisvag: I also like and use the builtin plugin installer, even outside of the bundle. I very frequently use it within a session, and love that it's been built to support in-session installations.  I would prefer warnings about potential breakages, or perhaps full on refusal to override certain packages (e.g. numpy/qt) in a pip-based installer, rather than wholesale removal.

Reply:
what if the plugin installer was a plugin? 🤔 

then it gets delivered as a separate Python package & bundled into the Desktop App

eases the maintenance burden on the core viewer that @jni highlights while enabling the added functionality for those who want it through a simple pip install

Reply:
Actually I like this idea, @neuromusic!

It could also open the possibility of having something to install plugins from "outside" napari (making it part of `napari-updater`), in case we need to repair an installation or simply customize things from outside. However, the plugin manager is Qt-based now, and maybe we don't want that for the napari-updater part (but 🤷 who knows). 

cc @goanpeca